test/request-template-mocha.js

```diff
@@ -0,0 +1,145 @@
+                network.graph.addInitial({port: 1337}, 'request', 'input');
+            });
+        }).should.become("Hello POST");
+    });
+    it("should support request headers", function() {
+        return createNetwork({
+            request: requestTemplate
+        }).then(function(network){
+            server.once('request', function(req, res) {
+                res.write("Hello " + req.headers['content-type']);
+                res.end();
+            });
+            var output = noflo.internalSocket.createSocket();
+            network.processes.request.component.outPorts.output.attach(output);
+            return new Promise(function(done) {
+                output.on('data', done);
+                network.graph.addInitial('POST', 'request', 'method');
+                network.graph.addInitial("http://localhost:{port}/", 'request', 'url');
+                network.graph.addInitial({'Content-Type': '{+type}'}, 'request', 'headers');
+                network.graph.addInitial({port: 1337, type: 'text/plain'}, 'request', 'input');
+            });
+        }).should.become("Hello text/plain");
+    });
+    it("should support request body", function() {
+        return createNetwork({
+            request: requestTemplate
+        }).then(function(network){
+            server.once('request', function(req, res) {
+                var body = [];
+                req.on('data', function(chunk) {
+                    body.push(chunk);
+                }).on('end', function() {
+                    body = Buffer.concat(body).toString();
+                    res.end(body);
+                });
+            });
+            var output = noflo.internalSocket.createSocket();
+            network.processes.request.component.outPorts.output.attach(output);
+            return new Promise(function(done) {
+                output.on('data', done);
+                network.graph.addInitial('POST', 'request', 'method');
+                network.graph.addInitial("http://localhost:{port}/", 'request', 'url');
+                network.graph.addInitial({'Content-Type': '{+type}'}, 'request', 'headers');
+                network.graph.addInitial('Hello {{{message}}}', 'request', 'body');
+                network.graph.addInitial({port: 1337, type: 'text/plain', message: "World"}, 'request', 'input');
+            });
+        }).should.become("Hello World");
+    });
+});
+
+/**
+ * Creates and starts a noflo.Network with a component for every component module
+ * given, however, no edges are present.
+ * Usage:
+ *  createNetwork({name:require('../components/rdf')}).then(function(network){
+ *      network.processes.name.component is the component instance
+ *      network.graph.addEdge('name', 'output', 'name', 'input') to add edge
+ *      network.graph.addInitial(data, 'name', 'input') to send data
+ *  });
+ */
+function createNetwork(componentModules) {
+    var graph = new noflo.Graph();
+    _.each(componentModules, function(module, name) {
```

Comment:
I'm fine with this being written this way, but I am curious to understand why.  Is this written using _.each (instead of 'for (name in componentModules)') because of JavaScript's lack of local variables?   Or because of the need to otherwise check for 'own' properties?  Or both?